src/main/java/de/tum/in/www1/artemis/service/connectors/ldap/LdapAuthenticationProvider.java

```diff
@@ -0,0 +1,180 @@
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.crypto.codec.Utf8;
+import org.springframework.security.ldap.SpringSecurityLdapTemplate;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.exception.ArtemisAuthenticationException;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.security.ArtemisAuthenticationProvider;
+import de.tum.in.www1.artemis.security.ArtemisAuthenticationProviderImpl;
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+import de.tum.in.www1.artemis.service.ldap.LdapUserDto;
+import de.tum.in.www1.artemis.service.ldap.LdapUserService;
+import de.tum.in.www1.artemis.service.user.AuthorityService;
+import de.tum.in.www1.artemis.service.user.PasswordService;
+import de.tum.in.www1.artemis.service.user.UserCreationService;
+import de.tum.in.www1.artemis.service.util.TimeLogUtil;
+
+@Component
+@Profile("ldap-only")
+@Primary
+@ComponentScan("de.tum.in.www1.artemis.*")
+public class LdapAuthenticationProvider extends ArtemisAuthenticationProviderImpl implements ArtemisAuthenticationProvider {
+
+    private final Logger log = LoggerFactory.getLogger(LdapAuthenticationProvider.class);
+
+    private final LdapUserService ldapUserService;
+
+    private final AuthorityService authorityService;
+
+    private final SpringSecurityLdapTemplate ldapTemplate;
+
+    public LdapAuthenticationProvider(UserRepository userRepository, LdapUserService ldapUserService, PasswordService passwordService, AuthorityService authorityService,
+            UserCreationService userCreationService, SpringSecurityLdapTemplate ldapTemplate) {
+        super(userRepository, passwordService, userCreationService);
+        this.ldapUserService = ldapUserService;
+        this.authorityService = authorityService;
+        this.ldapTemplate = ldapTemplate;
+    }
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        User user = getOrCreateUser(authentication, false);
+        if (user != null) {
+            return new UsernamePasswordAuthenticationToken(user.getLogin(), user.getPassword(), user.getGrantedAuthorities());
+        }
+        return null;
+    }
+
+    @Override
+    public User getOrCreateUser(Authentication authentication, String firstName, String lastName, String email, boolean skipPasswordCheck) {
+        // NOTE: firstName, lastName, email is not needed in this case since we always get these values from Jira
+        return getOrCreateUser(authentication, skipPasswordCheck);
+    }
+
+    private User getOrCreateUser(Authentication authentication, Boolean skipPasswordCheck) {
+        String username = authentication.getName().toLowerCase();
+        String password = authentication.getCredentials().toString();
```

Comment:
Teamscale clone warning, can you check if it's possible to elimate the clones?

Reply:
I looked into it and didn't find a way that would really make sense. `LdapAuthenticationProvider` and `JiraAuthentificationProvider` are in different subfolders and only have the class `ArtemisAuthenticationProviderImpl` in common from which they both inherit. We could move the few duplicated lines into the parent class, but that would mean `ArtemisInternalAuthenticationProvider` would also inherit it.